src/backend/BSSched/Checker.pm

```diff
@@ -702,7 +702,7 @@ sub expandandsort {
       $buildtype = 'modulemd';
     } elsif ($info && $info->{'file'}) {
       # directly implement most common types
-      if ($info->{'file'} =~ /\.(spec|dsc|kiwi|livebuild)$/) {
+      if ($info->{'file'} =~ /\.(spec|dsc|kiwi|livebuild)$/ || $info->{'file'} =~ m/^(mkosi)\./) {
```

Comment:
I think you should drop that change, as this is not that speed critical. There will probably not be tousands of mkosi recipes in the project, so we can just call the always correct Build::recipe2buildtype()

Reply:
@mlschroe thanks for your review, tested and removed.